---
title: "05.cyano_abundance_figures"
project: "Jech et al. Biocrust Cultivation"
author: "Sierra Jech"
lab: "Barger Lab"
date: "2024-10-24"
output: html_document
---

# Purpose
-  Create figures with points representing each mean absolute abundance and error bars

# Setup
Follows after running code in '02.phyloseq_ASV_Table.Rmd' and obtaining the processed phyloseq object. In addition, the '02.phyloseq_ASV_Table.Rmd' code produced a list of 265 cyanobacteria ASVs as a fasta file which was assigned correct names via Cydrasil and iTol. The new list is called 'data_output/cesucyanoassignments265.csv' and should be merged with cyano ASV dataframes when correct genera names need to be used. 

## Libraries

```{r}
library(phyloseq) # For all bioinformatics analyses post Dada2
#packageVersion('phyloseq')
library(ggplot2)
library(tidyverse)
library(vegan)
```

## Load Data
```{r}
cesu_16S_ps <- readRDS("data_output/mbcyano_processed_filt_ps.rds")
cyanoassignments <- read.csv("data/cesucyanoassignments265.csv")
cyanoassignments$GS <- paste(cyanoassignments$G_checked, cyanoassignments$S_checked)
qPCR <- read.csv("data/cesuqPCR.csv")
```

# Get to know the data, create labels, delete extra samples and blanks
```{r}
cesu_16S_ps@sam_data@names #colnames for the sample_data information in this phyloseq object
cesu_16S_ps@sam_data[["names"]] # view all samples

# Create a combo label where sample types are maintained but the replicate is not included (project, farm or greenhouse, desert, water, substrate)
cesu_16S_ps@sam_data$comboLabel <- paste(cesu_16S_ps@sam_data$project ,cesu_16S_ps@sam_data$farmGreenhouse, cesu_16S_ps@sam_data$desert, cesu_16S_ps@sam_data$water, cesu_16S_ps@sam_data$shade, cesu_16S_ps@sam_data$priorShade,cesu_16S_ps@sam_data$substrate, sep = ".")
unique(cesu_16S_ps@sam_data$comboLabel)
table(cesu_16S_ps@sam_data$comboLabel)

#Setup. Delete P and WP substrates, delete water regime B
# This means I can compare inocula growth in the greenhouse on no substrate, with and without shade (different shade types I think). I will only have timepoint 2 for mayberry farm. I have two timepoints for the greenhouse. I have control values at the greenhouse and farm at the end of cultivation. 

# Subsetting by sample name (remove all sods & watering regime B & blanks):
# now actually remove them in the ps object
cesu_16S_ps_subset <- subset_samples(cesu_16S_ps, !water %in% c("B"))
cesu_16S_ps_subset <- subset_samples(cesu_16S_ps_subset, !substrate %in% c("P", "WP"))
unique(cesu_16S_ps_subset@sam_data$comboLabel)
table(cesu_16S_ps_subset@sam_data$comboLabel)
# one of the blanks has an incorrect value for substrate and shade so it is separating out from the rest in the combolabels - fix it when you melt
# Contruct other useful labels?
# remove blanks for now
cesu_16S_ps_subset <- subset_samples(cesu_16S_ps_subset, !project %in% c("Blank"))

# check it
cesu_16S_ps_subset@sam_data[["comboLabel"]] # ok
# the initials are still in there

# factor the label levels for plotting
cesu_16S_ps_subset@sam_data$comboLabel <- factor(cesu_16S_ps_subset@sam_data$comboLabel, levels = c("CP.NA.initial.NA.none.NA.soil", "CP.NA.initial.NA.none.none.soil", "MJ.NA.initial.NA.none.NA.soil", "GH0.NA.CP.3.none.NA.greenhouseSoil", "GH0.NA.CP.3.S.NA.greenhouseSoil", "GH0.NA.MJ.3.none.NA.greenhouseSoil", "GH0.NA.MJ.3.S.NA.greenhouseSoil", "GH1.NA.control.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil", "GH1.NA.CP.3.none.NA.greenhouseSoil", "GH1.NA.CP.3.S.NA.greenhouseSoil", "GH1.NA.MJ.3.none.NA.greenhouseSoil", "GH1.NA.MJ.3.S.NA.greenhouseSoil", "MB2.NA.C.A.none.NA.N", "MB2.NA.C.A.S.NA.JC", "MB2.NA.CP.A.none.NA.N", "MB2.NA.CP.A.S.NA.JC", "MB2.NA.MJ.A.none.NA.N", "MB2.NA.MJ.A.S.NA.JC", "R0.g.CP.NA.NA.none.soil", "R0.g.CP.NA.NA.S.soil", "R0.g.MJ.NA.NA.none.soil", "R0.f.MJ.NA.NA.none.soil", "R0.g.MJ.NA.NA.S.soil", "R0.f.CP.NA.NA.none.soil", "R0.f.CP.NA.NA.S.soil", "R0.f.MJ.NA.NA.S.soil", "R0.control.control.NA.NA.none.soil", "R2.g.CP.NA.NA.none.soil", "R2.f.CP.NA.NA.none.soil", "R2.g.CP.NA.NA.S.soil", "R2.f.CP.NA.NA.S.soil", "R2.g.MJ.NA.NA.none.soil", "R2.f.MJ.NA.NA.none.soil", "R2.g.MJ.NA.NA.S.soil", "R2.f.MJ.NA.NA.S.soil", "R2.control.control.NA.NA.none.soil"))

cesu_16S_ps_subset@sam_data$desert <- factor(cesu_16S_ps_subset@sam_data$desert, levels = c("control", "C", "CP", "MJ"))


#Get count of total reads to calculate relative abundance
table(phyloseq::tax_table(cesu_16S_ps_subset)[, "Phylum"]) #40 Phyla included
# get the total read count per sample
reads_per_sample <- as.data.frame (sample_sums(cesu_16S_ps_subset)) # they are all different because we did not rarefy in this dataset
#copy the rownames to a column
reads_per_sample <- tibble::rownames_to_column(reads_per_sample, "Sample")
# change the name of the second column
colnames(reads_per_sample) <- c("Sample", "totalReads")
nrow(reads_per_sample) # 143 samples are included


# melt the data out of phyloseq format and into long dataframe
cesu_psmelt <- psmelt(cesu_16S_ps_subset) #this takes some time, patience!

# In order to create a plot that displays the cyano average relative abundance for all treatments, I need to subset the dataframe for all the cyanos, calculate their relative abundance, then plot. I might as well create dataframes that can be used to display other phyla as well, so we will do the top 10

# Get a list of the most abundant Phyla
temp <- cesu_psmelt %>%
  group_by(Phylum) %>%
  summarise(totalAbund = sum(Abundance)) %>%
  arrange(-totalAbund)

temp <- temp[1:10,]

topPhyla <- temp$Phylum # list that holds the TopPhyla names
topPhyla

#subset the psmelt dataframe for only the TopPhyla
phylaSubset <- cesu_psmelt %>%
  filter(Phylum %in% topPhyla)

# check that we got the right ones
unique(phylaSubset$Phylum) # yes

# calculate the reads per Phylum, grouped by Sample
phylaSubset_abund <- phylaSubset %>%
  group_by(Sample, Phylum) %>%
  summarise(abund = sum(Abundance))

#make a new column were I divide each of the values by the correct one in the reads_per_sample list
# first merge the reads_per_sample data onto the dataframe
phylaSubset_abund2 <- inner_join(phylaSubset_abund, reads_per_sample, by = "Sample")
# calculate relative abundance: divide the abund column by the total reads column
phylaSubset_abund2$relAbund <- phylaSubset_abund2$abund / phylaSubset_abund2$totalReads
# check that it does not add to 1 because there should be some in the Other category
total_relAbund <- phylaSubset_abund2 %>%
  group_by(Sample) %>%
  summarise(sum = sum(relAbund))
head(total_relAbund)
# add a new column which contains the relative abundance value for the Other category
total_relAbund$other <- (1 - total_relAbund$sum)
# delete the sum column
total_relAbund <- total_relAbund %>%
  dplyr::select(Sample, other)
#add column which is "Other" repeated
total_relAbund$Phylum <- "Other"
#change column header "other" to relAbund
colnames(total_relAbund)[which(names(total_relAbund) == "other")] <- "relAbund"
# select columns to keep in the dataframe we want
phylaSubset_abund2 <- phylaSubset_abund2 %>%
  dplyr::select(Sample, relAbund, Phylum)
# rbind the other values to the phylaSubset_abund2 dataframe
phylaSubset_abund3 <- rbind(phylaSubset_abund2, total_relAbund)

# Now check that they sum to 1
total_relAbund2 <- phylaSubset_abund3 %>%
  group_by(Sample) %>%
  summarise(sum = sum(relAbund)) 
head(total_relAbund2) # YES!!!!

# Plot Relative Abundance
temp2 <- unique(cesu_psmelt %>% dplyr::select(Sample, comboLabel, desert, project, shade))
phylaSubset_abund3 <- left_join(phylaSubset_abund3, temp2, by = "Sample")

# Plot Relative Abundance - basic first glimpse
# phylaSubset_abund3 %>%
#   ggplot(aes(y = relAbund, x = Sample, fill = Phylum)) +
#   geom_bar(stat = "identity", colour = "black", linewidth = 0.25) +
#   labs(x = "", y = "Relative Abundance", fill = "Phyla") +
#   theme_classic()+
#   coord_cartesian(ylim = c(0,1), expand = FALSE)+
#   theme(axis.title = element_text(face = "bold", size = 16),
#         axis.text.x = element_text(size = 12, angle = 90),
#         axis.text.y = element_text(size = 12),
#         legend.text = element_text(face = "italic"),
#         axis.ticks.x = element_blank())

# All together but average (not every replicate)
# first calculate the averages
# CP has two different comboLabels for the initial samples. Fix that
unique(phylaSubset_abund3$comboLabel)
# replace 
phylaSubset_abund3$comboLabel[phylaSubset_abund3$comboLabel == "CP.NA.initial.NA.none.none.soil"] <- "CP.NA.initial.NA.none.NA.soil"
unique(phylaSubset_abund3$comboLabel) # it worked!


# now aggregate at Phylum level but keep important grouping variables
phylaSubset_abund3_agregate <- phylaSubset_abund3 %>% 
  group_by(comboLabel, Phylum, desert, shade, project) %>%
  summarise(meanRel_abund = mean(relAbund),
            sdRel_abund = sd(relAbund))

unique(phylaSubset_abund3_agregate$comboLabel)

unique(phylaSubset_abund3_agregate$Phylum) # need 11 colors

#this factoring isn't needed because it is wrong.
#phylaSubset_abund3_agregate$comboLabel <- factor(phylaSubset_abund3_agregate$comboLabel, levels = c("CP.NA.initial.NA.none.NA.soil", "MJ.NA.initial.NA.none.NA.soil", "GH0.NA.CP.3.none.NA.greenhouseSoil", "GH0.NA.CP.3.S.NA.greenhouseSoil", "GH0.NA.MJ.3.none.NA.greenhouseSoil", "GH0.NA.MJ.3.S.NA.greenhouseSoil", "GH1.NA.control.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil", "GH1.NA.CP.3.none.NA.greenhouseSoil", "GH1.NA.CP.3.S.NA.greenhouseSoil", "GH1.NA.MJ.3.none.NA.greenhouseSoil", "GH1.NA.MJ.3.S.NA.greenhouseSoil", "MB2.NA.C.A.none.NA.N", "MB2.NA.C.A.S.NA.JC", "MB2.NA.CP.A.none.NA.N", "MB2.NA.CP.A.S.NA.JC", "MB2.NA.MJ.A.none.NA.N", "MB2.NA.MJ.A.S.NA.JC", "R0.g.CP.NA.NA.none.soil", "R0.g.CP.NA.NA.S.soil", "R0.g.MJ.NA.NA.none.soil", "R0.f.MJ.NA.NA.none.soil", "R0.g.MJ.NA.NA.S.soil", "R0.f.CP.NA.NA.none.soil", "R0.f.CP.NA.NA.S.soil", "R0.f.MJ.NA.NA.S.soil", "R0.control.control.NA.NA.none.soil", "R2.g.CP.NA.NA.none.soil", "R2.f.CP.NA.NA.none.soil", "R2.g.CP.NA.NA.S.soil", "R2.f.CP.NA.NA.S.soil", "R2.g.MJ.NA.NA.none.soil", "R2.f.MJ.NA.NA.none.soil", "R2.g.MJ.NA.NA.S.soil", "R2.f.MJ.NA.NA.S.soil", "R2.control.control.NA.NA.none.soil"))

# plot
#new_labels <- c("CPinitial" = "Colorado\nPlateau", "MJinitial" = "Mojave\nDesert","SNinitial" = "Sonoran\nDesert")

# get the order for the topPhyla
topPhyla

phylaSubset_abund3_agregate$Phylum <- factor(phylaSubset_abund3_agregate$Phylum, levels = c("Other","Abditibacteriota","Planctomycetota","Verrucomicrobiota","Crenarchaeota", "Chloroflexi", "Actinobacteriota", "Acidobacteriota","Cyanobacteria", "Proteobacteria","Bacteroidota"))

# in stacked bar plots and facets, the y axis is plotted backwards, so I need to do some rearranging of the levels. This factoring is for both CP and MJ 
phylaSubset_abund3_agregate$comboLabel2 <- phylaSubset_abund3_agregate$comboLabel
phylaSubset_abund3_agregate$comboLabel2 <- factor(phylaSubset_abund3_agregate$comboLabel2, levels = c("CP.NA.initial.NA.none.NA.soil", "MJ.NA.initial.NA.none.NA.soil", "GH0.NA.CP.3.S.NA.greenhouseSoil", "GH0.NA.CP.3.none.NA.greenhouseSoil", "GH0.NA.MJ.3.S.NA.greenhouseSoil","GH0.NA.MJ.3.none.NA.greenhouseSoil",  "GH1.NA.CP.3.S.NA.greenhouseSoil", "GH1.NA.CP.3.none.NA.greenhouseSoil", "GH1.NA.MJ.3.S.NA.greenhouseSoil", "GH1.NA.MJ.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil", "GH1.NA.control.3.none.NA.greenhouseSoil", "MB2.NA.CP.A.S.NA.JC", "MB2.NA.CP.A.none.NA.N", "MB2.NA.MJ.A.S.NA.JC", "MB2.NA.MJ.A.none.NA.N","MB2.NA.C.A.S.NA.JC", "MB2.NA.C.A.none.NA.N", "R0.f.CP.NA.NA.S.soil", "R0.f.CP.NA.NA.none.soil","R0.g.CP.NA.NA.S.soil", "R0.g.CP.NA.NA.none.soil","R0.f.MJ.NA.NA.S.soil", "R0.f.MJ.NA.NA.none.soil", "R0.g.MJ.NA.NA.S.soil","R0.g.MJ.NA.NA.none.soil","R0.control.control.NA.NA.none.soil","R2.f.CP.NA.NA.S.soil","R2.f.CP.NA.NA.none.soil", "R2.g.CP.NA.NA.S.soil","R2.g.CP.NA.NA.none.soil","R2.f.MJ.NA.NA.S.soil", "R2.f.MJ.NA.NA.none.soil", "R2.g.MJ.NA.NA.S.soil", "R2.g.MJ.NA.NA.none.soil","R2.control.control.NA.NA.none.soil"))
```

# Colorado Plateau Plot
```{r}
new_labels_CP <- c("CP.NA.initial.NA.none.NA.soil" = "Reference biocrust",
                   "GH0.NA.CP.3.none.NA.greenhouseSoil" = "No shade",
                   "GH0.NA.CP.3.S.NA.greenhouseSoil" = "Shade",
                   "GH1.NA.control.3.none.NA.greenhouseSoil" = "No inoculum, no shade",
                   "GH1.NA.control.3.S.NA.greenhouseSoil" = "No inoculum, shade",
                   "GH1.NA.CP.3.none.NA.greenhouseSoil" = "No shade",
                   "GH1.NA.CP.3.S.NA.greenhouseSoil" = "Shade",
                   "MB2.NA.C.A.none.NA.N" = "No inoculum, no shade",
                   "MB2.NA.C.A.S.NA.JC" = "No inoculum, shade",
                   "MB2.NA.CP.A.none.NA.N" = "No shade",
                   "MB2.NA.CP.A.S.NA.JC" = "Shade",
                   "R0.g.CP.NA.NA.none.soil" = "Greenhouse cultivated, no shade",
                   "R0.g.CP.NA.NA.S.soil" = "Greenhouse cultivated, shade",
                   "R0.f.CP.NA.NA.none.soil" = "Field cultivated, no shade",
                   "R0.f.CP.NA.NA.S.soil" = "Field cultivated, shade",
                   "R0.control.control.NA.NA.none.soil" = "No inoculum",
                   "R2.g.CP.NA.NA.none.soil" = "Greenhouse cultivated, no shade",
                   "R2.f.CP.NA.NA.none.soil" = "Field cultivated, no shade",
                   "R2.g.CP.NA.NA.S.soil" = "Greenhouse cultivated, shade",
                   "R2.f.CP.NA.NA.S.soil" = "Field cultivated, shade",
                   "R2.control.control.NA.NA.none.soil" = "No inoculum")

project_labels <- c("CP" = "", "MJ" = "", "GH0" = "Greenhouse\ncultivation\ninitial", "GH1" = "Greenhouse\ncultivation\nend point","MB2" = "Field\ncultivation\nend point", "R0" = "Restoration\ninitial", "R2"= "Restoration\nend point")

levels(phylaSubset_abund3_agregate$project)
phylaSubset_abund3_agregate$project <- factor(phylaSubset_abund3_agregate$project, levels = c("CP", "MJ", "GH0", "GH1","MB2","R0","R2"))

ten_colors_green_remixed <- c('#536A89', '#BE6C9C','#658354', '#0F74C8', '#4E467A', '#9591BF','#2A659B','#AB3A62', '#508EBF',"darkgreen",'#E395C2')

phylaSubset_abund3_agregate %>%
  filter(comboLabel2 %in% c("CP.NA.initial.NA.none.NA.soil", "GH0.NA.CP.3.none.NA.greenhouseSoil", "GH0.NA.CP.3.S.NA.greenhouseSoil", "GH1.NA.control.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil", "GH1.NA.CP.3.none.NA.greenhouseSoil", "GH1.NA.CP.3.S.NA.greenhouseSoil", "MB2.NA.C.A.none.NA.N", "MB2.NA.C.A.S.NA.JC", "MB2.NA.CP.A.none.NA.N", "MB2.NA.CP.A.S.NA.JC", "R0.g.CP.NA.NA.none.soil", "R0.g.CP.NA.NA.S.soil", "R0.f.CP.NA.NA.none.soil", "R0.f.CP.NA.NA.S.soil", "R0.control.control.NA.NA.none.soil", "R2.g.CP.NA.NA.none.soil", "R2.f.CP.NA.NA.none.soil", "R2.g.CP.NA.NA.S.soil", "R2.f.CP.NA.NA.S.soil","R2.control.control.NA.NA.none.soil")) %>%
  ggplot(aes(y = meanRel_abund, x = comboLabel2, fill = Phylum)) +
  geom_bar(stat = "identity", alpha = 0.8, width = 0.85) +
  geom_col(color = "black", linewidth = 0.25)+
  scale_fill_manual(values = ten_colors_green_remixed, breaks=c("Bacteroidota", "Proteobacteria","Cyanobacteria", "Acidobacteriota", "Actinobacteriota", "Chloroflexi", "Crenarchaeota", "Verrucomicrobiota", "Planctomycetota", "Abditibacteriota",'Other'))+
  scale_y_continuous(expand = c(0, 0.01), breaks=seq(0, 1, .2)) +
  scale_x_discrete(labels = new_labels_CP)+
  theme_classic()+
  coord_flip()+
  facet_grid(rows = vars(project), scales = "free", space = "free", labeller = labeller(project = project_labels))+
  labs(x = "", y = "Relative Abundance", fill = "")+
  theme(axis.text = element_text(size = 12),
        #axis.text.y = element_blank(),
        #axis.ticks.y = element_blank(),
        axis.title = element_text(size = 12),
        strip.background = element_rect(fill="white", colour=NA),
        #strip.background = element_blank(),
        #strip.text = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(face="bold", size=12),
        strip.text.y.right = element_text(angle = 0),
        legend.title = element_text(face = "bold"),
        legend.position = "bottom",
        legend.text = element_text(size = 10),
        plot.margin = margin(0.5,1,0,0, "cm"))+
  guides(fill = guide_legend(nrow = 3, byrow = TRUE))

# include this plot in the supplementary information
#ggsave("figures/allCPsamples_phyla_relAbund_stackedbar.pdf", width = 10, height = 9)
```

# And again for Mojave
```{r}
new_labels_MJ <- c("MJ.NA.initial.NA.none.NA.soil" = "Reference biocrust",
                   "GH0.NA.MJ.3.none.NA.greenhouseSoil" = "No shade",
                   "GH0.NA.MJ.3.S.NA.greenhouseSoil" = "Shade",
                   "GH1.NA.MJ.3.none.NA.greenhouseSoil" = "No shade",
                   "GH1.NA.MJ.3.S.NA.greenhouseSoil" = "Shade",
                   "GH1.NA.control.3.none.NA.greenhouseSoil" = "No inoculum, no shade",
                   "GH1.NA.control.3.S.NA.greenhouseSoil" = "No inoculum, shade",
                   "MB2.NA.MJ.A.none.NA.N" = "No shade",
                   "MB2.NA.MJ.A.S.NA.JC" = "Shade",
                   "MB2.NA.C.A.none.NA.N" = "No inoculum, no shade",
                   "MB2.NA.C.A.S.NA.JC" = "No inoculum, shade",
                   "R0.g.MJ.NA.NA.none.soil"= "Greenhouse cultivated, no shade",
                   "R0.f.MJ.NA.NA.none.soil" = "Field cultivated, no shade",
                   "R0.g.MJ.NA.NA.S.soil" = "Greenhouse cultivated, shade",
                   "R0.f.MJ.NA.NA.S.soil" = "Field cultivated, shade",
                   "R0.control.control.NA.NA.none.soil" = "No inoculum",
                   "R2.g.MJ.NA.NA.none.soil" = "Greenhouse cultivated, no shade",
                   "R2.f.MJ.NA.NA.none.soil" = "Field cultivated, no shade",
                   "R2.g.MJ.NA.NA.S.soil" = "Greenhouse cultivated, shade",
                   "R2.f.MJ.NA.NA.S.soil" = "Field cultivated, shade",
                   "R2.control.control.NA.NA.none.soil" = "No inoculum"
)

phylaSubset_abund3_agregate %>%
  filter(comboLabel2 %in% c("MJ.NA.initial.NA.none.NA.soil", "GH0.NA.MJ.3.none.NA.greenhouseSoil", "GH0.NA.MJ.3.S.NA.greenhouseSoil", "GH1.NA.control.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil", "GH1.NA.MJ.3.none.NA.greenhouseSoil", "GH1.NA.MJ.3.S.NA.greenhouseSoil", "MB2.NA.C.A.none.NA.N", "MB2.NA.C.A.S.NA.JC", "MB2.NA.MJ.A.none.NA.N", "MB2.NA.MJ.A.S.NA.JC", "R0.g.MJ.NA.NA.none.soil", "R0.g.MJ.NA.NA.S.soil", "R0.f.MJ.NA.NA.none.soil", "R0.f.MJ.NA.NA.S.soil", "R0.control.control.NA.NA.none.soil", "R2.g.MJ.NA.NA.none.soil", "R2.f.MJ.NA.NA.none.soil", "R2.g.MJ.NA.NA.S.soil", "R2.f.MJ.NA.NA.S.soil","R2.control.control.NA.NA.none.soil")) %>%
  ggplot(aes(y = meanRel_abund, x = comboLabel2, fill = Phylum)) +
  geom_bar(stat = "identity", alpha = 0.8, width = 0.85) +
  geom_col(color = "black", linewidth = 0.25)+
  scale_fill_manual(values = ten_colors_green_remixed, breaks=c("Bacteroidota", "Proteobacteria","Cyanobacteria", "Acidobacteriota", "Actinobacteriota", "Chloroflexi", "Crenarchaeota", "Verrucomicrobiota", "Planctomycetota", "Abditibacteriota",'Other'))+
  scale_y_continuous(expand = c(0, 0.01), breaks=seq(0, 1, .2)) +
  scale_x_discrete(labels = new_labels_MJ)+
  theme_classic()+
  coord_flip()+
  facet_grid(rows = vars(project), scales = "free", space = "free", labeller = labeller(project = project_labels))+
  labs(x = "", y = "Relative Abundance", fill = "")+
  theme(axis.text = element_text(size = 12),
        #axis.text.y = element_blank(),
        #axis.ticks.y = element_blank(),
        axis.title = element_text(size = 12),
        strip.background = element_rect(fill="white", colour=NA),
        #strip.background = element_blank(),
        #strip.text = element_blank(),
        strip.text = element_text(face="bold", size=12),
        strip.text.y.right = element_text(angle = 0),
        legend.title = element_text(face = "bold"),
        legend.position = "bottom",
        legend.text = element_text(size = 10),
        plot.margin = margin(0.5,1,0,0, "cm"))+
  guides(fill = guide_legend(nrow = 3, byrow = TRUE))

# include this plot in the supplementary information
#ggsave("figures/allMJsamples_phyla_relAbund_stackedbar.pdf", width = 10, height = 9)
```

Now that I have phyla-level relative abundance information, I can plot the cyano abundances over time. 
```{r}
# calculate the average value for CP and MJ reference crusts. Use the dataset that contains relative abundance for each sample at the phylum level
intact_mean_phyla_rel_abund <- phylaSubset_abund3 %>% 
  filter(project %in% c("CP", "MJ")) %>% 
  group_by(project, Phylum) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me values for CP and MJ reference biocrusts


# calculate the greenhouse cultivation initial value. Use the same dataset. WE ARE NOT USING THESE VALUES BUT THEY ARE SIMILAR TO THE CALCULATED STATIC COMMUNITY VALUES
# ghInitial_mean_phyla_rel_abund <- phylaSubset_abund3 %>%
#   filter(comboLabel %in% c("GH0.NA.CP.3.none.NA.greenhouseSoil", "GH0.NA.MJ.3.none.NA.greenhouseSoil", "GH0.NA.CP.3.S.NA.greenhouseSoil", "GH0.NA.MJ.3.S.NA.greenhouseSoil")) %>%
#   group_by(desert,Phylum) %>% 
#   summarise(meanRelAbund = mean(relAbund),
#             sdRelAbund = sd(relAbund)) # gives me starting values in the greenhouse for both CP and MJ inoculations

# calculate the greenhouse cultivation no inoculum values. Use the same dataset
ghNoInoculum_mean_phyla_rel_abund <- phylaSubset_abund3 %>%
  filter(comboLabel %in% c("GH1.NA.control.3.none.NA.greenhouseSoil", "GH1.NA.control.3.S.NA.greenhouseSoil")) %>%
  group_by(shade, desert, Phylum) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me no inoculum (control) values in the greenhouse

# calculate the greenhouse static community value. Use the already calculated values (Cyanobacteria)
# value of cyanos in CP reference biocrusts = 0.245
# value of cyanos in MJ reference biocrusts = 0.581
# value of cyanos in uninoculated soil in the greenhouse = 0.052
# CP greenhouse calculated static Cyano
(0.2*0.245)+(0.8*0.052) # 0.091
# MJ greenhouse calculated static Cyano
(0.2*0.581)+(0.8*0.052) # 0.158

# get values for shaded and unshaded (both deserts) greenhouse cultivation
gh_cult_phyla_relAbund <- phylaSubset_abund3 %>%
  filter(comboLabel %in% c("GH1.NA.CP.3.none.NA.greenhouseSoil", "GH1.NA.MJ.3.none.NA.greenhouseSoil", "GH1.NA.CP.3.S.NA.greenhouseSoil", "GH1.NA.MJ.3.S.NA.greenhouseSoil")) %>%
  group_by(shade, desert, Phylum) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me no inoculum (control) values in the greenhouse
```

# Greenhouse Cultivation Plot = CP
```{r}
GHcolors <- c("darkseagreen", "darkslategrey")

gh_cult_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("CP")) %>%
  ggplot(mapping = aes(x = shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 2.5, ymin = 0.245-0.023, ymax=0.245+0.023), fill = "grey", alpha = 0.4)+
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.052, yend = 0.052),linetype = "dashed")+ #no inoculum, no shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.265, yend = 0.265))+ #no inoculum, shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.091, yend = 0.091), linetype = "dotted")+ #calculated static community
  geom_point(mapping = aes(color = shade), size = 3)+
  geom_errorbar(mapping = aes(x = shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = shade), width = 0.05)+
  scale_x_discrete(labels = c("none" = "", "S" = ""))+
  scale_color_manual(values = GHcolors, guide = "none")+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  theme(axis.text = element_text(size = 11),
        axis.title = element_text(size = 11, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank())+
  ylim(0,1)

#ggsave("figures/CPcyano_relAbund_dots_GHcult.pdf", width = 3, height = 3)
```

# Greenhouse Cultivation Plot = MJ
```{r}
gh_cult_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("MJ")) %>%
  ggplot(mapping = aes(x = shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 2.5, ymin = 0.581-0.02, ymax=0.581+0.02), fill = "grey", alpha = 0.4)+
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.052, yend = 0.052), linetype = "dashed")+ #no inoculum, no shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.265, yend = 0.265))+ #no inoculum, shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.158, yend = 0.158), linetype = "dotted")+ #calculated static community
  geom_point(mapping = aes(color = shade), size = 3)+
  geom_errorbar(mapping = aes(x = shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = shade), width = 0.05)+
  scale_x_discrete(labels = c("none" = "", "S" = ""))+
  scale_color_manual(values = GHcolors, guide = "none")+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  theme(axis.text = element_text(size = 11),
        axis.title = element_text(size = 11, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank())+
  ylim(0,1)
  
#ggsave("figures/MJcyano_relAbund_dots_GHcult.pdf", width = 3, height = 3)
```

# FIELD Cultivation Plots
```{r}
Fieldcolors <- c("darksalmon", "brown4")

# calculate the average value for CP background soil. Use the same dataset
bareSoil_mean_phyla_rel_abund <- phylaSubset_abund3 %>%
  filter(comboLabel %in% c("R2.control.control.NA.NA.none.soil", "R0.control.control.NA.NA.none.soil")) %>%
  group_by(Phylum) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me bare soil values on the Colorado Plateau

# calculate the average value for CP no inouclum in the field. Use the same dataset
fieldNoInoculum_mean_phyla_rel_abund <- phylaSubset_abund3 %>%
  filter(comboLabel %in% c("MB2.NA.C.A.none.NA.N","MB2.NA.C.A.S.NA.JC")) %>%
  group_by(Phylum,shade) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me bare soil values on the Colorado Plateau


# calculate the field static community value. Use the already calculated values (Cyanobacteria)
# value of cyanos in CP reference biocrusts = 0.245
# value of cyanos in MJ reference biocrusts = 0.581
# value of cyanos in uninoculated soil in the field = 0.014
# CP field calculated static Cyano
(0.2*0.245)+(0.8*0.012) # 0.0586
# MJ greenhouse calculated static Cyano
(0.2*0.581)+(0.8*0.012) # 0.126

# get values for shaded and unshaded (both deserts) field cultivation (MB2)
field_cult_phyla_relAbund <- phylaSubset_abund3 %>%
  filter(comboLabel %in% c("MB2.NA.CP.A.none.NA.N", "MB2.NA.MJ.A.none.NA.N", "MB2.NA.MJ.A.S.NA.JC", "MB2.NA.CP.A.S.NA.JC")) %>%
  group_by(shade, desert, Phylum) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # gives me no inoculum (control) values in the greenhouse

# Field Cultivation Plot = CP
field_cult_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("CP")) %>%
  ggplot(mapping = aes(x = shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 2.5, ymin = 0.245-0.023, ymax=0.245+0.023), fill = "grey", alpha = 0.4)+
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.012, yend = 0.012),linetype = "dashed")+ #no inoculum, no shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.255, yend = 0.255))+ #no inoculum, shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.06, yend = 0.06), linetype = "dotted")+ #calculated static community
  geom_point(mapping = aes(color = shade), size = 3)+
  geom_errorbar(mapping = aes(x = shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = shade), width = 0.05)+
  scale_x_discrete(labels = c("none" = "", "S" = ""))+
  scale_color_manual(values = Fieldcolors, guide = "none")+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  theme(axis.text = element_text(size = 11),
        axis.title = element_text(size = 11, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank())+
  ylim(0,1)

#ggsave("figures/CPcyano_relAbund_dots_fieldcult.pdf", width = 3, height = 3)
```

# Field Cultivation Plot = MJ
```{r}
field_cult_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("MJ")) %>%
  ggplot(mapping = aes(x = shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 2.5, ymin = 0.581-0.02, ymax=0.581+0.02), fill = "grey", alpha = 0.4)+
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.012, yend = 0.012), linetype = "dashed")+ #no inoculum, no shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.265, yend = 0.265))+ #no inoculum, shade
  geom_segment(aes(x = 0.5, xend = 2.5, y = 0.126, yend = 0.126), linetype = "dotted")+ #calculated static community
  geom_point(mapping = aes(color = shade), size = 3)+
  geom_errorbar(mapping = aes(x = shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = shade), width = 0.05)+
  scale_x_discrete(labels = c("none" = "", "S" = ""))+
  scale_color_manual(values=Fieldcolors, guide = "none")+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  theme(axis.text = element_text(size = 11),
        axis.title = element_text(size = 11, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank())+
  ylim(0,1)

#ggsave("figures/MJcyano_relAbund_dots_fieldcult.pdf", width = 3, height = 3)
```

# Now the same plots but for the restoration data 
Use the background soils as the baseline, use the intact biocrust as the target, use the R0 data as the initial starting (so there will be two time points on this chart) and 4 conditions (GH, GHam, Field, Fieldam)

```{r}
bareSoil_mean_phyla_rel_abund %>% 
  filter(Phylum %in% c("Cyanobacteria")) # mean = 0.138, sd = 0.0152
intact_mean_phyla_rel_abund %>%
  filter(Phylum %in% c("Cyanobacteria")) # mean CP = 0.245, sd = 0.0226
# mean MJ = 0.581, sd = 0.019

# calculate values for restoration (8 means)
temp3 <- unique(cesu_psmelt %>% dplyr::select(Sample, priorShade, farmGreenhouse))
  
phylaSubset_abund4 <- left_join(phylaSubset_abund3, temp3, by = "Sample")

resto_phyla_relAbund <- phylaSubset_abund4 %>%
  filter(comboLabel %in% c("R0.g.CP.NA.NA.none.soil","R0.g.MJ.NA.NA.none.soil", "R0.f.CP.NA.NA.none.soil", "R0.f.MJ.NA.NA.none.soil", "R0.g.CP.NA.NA.S.soil", "R0.g.MJ.NA.NA.S.soil", "R0.f.CP.NA.NA.S.soil", "R0.f.MJ.NA.NA.S.soil", "R2.g.CP.NA.NA.none.soil", "R2.g.MJ.NA.NA.none.soil", "R2.f.CP.NA.NA.none.soil", "R2.f.MJ.NA.NA.none.soil", "R2.g.CP.NA.NA.S.soil", "R2.g.MJ.NA.NA.S.soil", "R2.f.CP.NA.NA.S.soil", "R2.f.MJ.NA.NA.S.soil")) %>%
  group_by(Phylum, desert, project, priorShade, farmGreenhouse) %>% 
  summarise(meanRelAbund = mean(relAbund),
            sdRelAbund = sd(relAbund)) # missing f/g information and prior shade information

# make a new label for plotting
resto_phyla_relAbund$farmGreenhouse_shade <- paste(resto_phyla_relAbund$farmGreenhouse, resto_phyla_relAbund$priorShade)

resto_phyla_relAbund$farmGreenhouse_shade <- factor(resto_phyla_relAbund$farmGreenhouse_shade, levels = c("g none", "g S", "f none", "f S"))
```

# Restoration Trials Plot = CP
```{r}
restoColors <- c("darkseagreen", "darkslategrey","darksalmon", "brown4")

resto_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("CP")) %>%
  ggplot(mapping = aes(x = farmGreenhouse_shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 4.5, ymin = 0.245-0.023, ymax=0.245+0.023), fill = "grey", alpha = 0.4)+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 4.5, ymin = 0.138-0.015, ymax=0.138+0.015), fill = "grey90", alpha = 0.4)+ # bare soil, no inoculum 
  #geom_segment(aes(x = 0.5, xend = 4.5, y = 0.138, yend = 0.138), linetype = "dashed")+ # bare soil, no inoculum 
  geom_point(mapping = aes(color = farmGreenhouse_shade, shape = project, alpha = project), size = 3)+
  geom_errorbar(mapping = aes(x = farmGreenhouse_shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = farmGreenhouse_shade, alpha = project), width = 0.05)+
  scale_x_discrete(labels = c("g none" = "", "g S" = "", "f none" = "", "f S" = ""))+
  scale_color_manual(values=restoColors, guide = "none")+
  scale_alpha_discrete(range = c(0.5, 1), labels = c("R0" = "Start", "R2" = "End"))+
  scale_shape_manual(values = c(1, 16), labels = c("R0" = "Start", "R2" = "End"))+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  guides(shape = guide_legend(title = "Restoration"),
         alpha = guide_legend(title = "Restoration"))+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.position = "bottom",
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12))+
  ylim(-0.05,1)
  
#ggsave("figures/CPrestocyano_relAbund_dots.pdf", width = 7, height = 4)
```

# Restoration Trials = MJ
```{r}
resto_phyla_relAbund %>% 
  filter(Phylum %in% c("Cyanobacteria")) %>%
  filter(desert %in% c("MJ")) %>%
  ggplot(mapping = aes(x = farmGreenhouse_shade, y = meanRelAbund))+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 4.5, ymin = 0.581-0.019, ymax=0.581+0.019), fill = "grey", alpha = 0.4)+
  geom_rect(mapping = aes(xmin = 0.5, xmax = 4.5, ymin = 0.138-0.015, ymax=0.138+0.015), fill = "grey90", alpha = 0.4)+ # bare soil, no inoculum 
  #geom_segment(aes(x = 0.5, xend = 4.5, y = 0.138, yend = 0.138), linetype = "dashed")+ # bare soil, no inoculum 
  geom_point(mapping = aes(color = farmGreenhouse_shade, shape = project, alpha = project), size = 3)+
  geom_errorbar(mapping = aes(x = farmGreenhouse_shade, ymin = meanRelAbund-sdRelAbund, ymax = meanRelAbund+sdRelAbund, color = farmGreenhouse_shade, alpha = project), width = 0.05)+
  scale_x_discrete(labels = c("g none" = "", "g S" = "", "f none" = "", "f S" = ""))+
  scale_color_manual(values=restoColors, guide = "none")+
  scale_alpha_discrete(range = c(0.5, 1), labels = c("R0" = "Start", "R2" = "End"))+
  scale_shape_manual(values = c(1, 16), labels = c("R0" = "Start", "R2" = "End"))+
  labs(x = "", y = "Cyanobacteria Relative Abundance")+
  theme_classic()+
  guides(shape = guide_legend(title = "Restoration"),
         alpha = guide_legend(title = "Restoration"))+
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 12, face = "bold"),
        axis.line.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12),
        legend.position = "bottom")+
  ylim(-0.05,1)

#ggsave("figures/MJrestocyano_relAbund_dots.pdf", width = 7, height = 4)
```